File: 卡码/代码随想录/Python/链表/05-删除链表倒数第n个节点/LeetCode_0019.py
#19. 删除链表的倒数第 N 个结点
from typing import *

class ListNode:
	def __init__(self,val=0,next=None):
		self.val=val
		self.next=next

class Solution:
	def removeNthFromEnd(self,head:Optional[ListNode],n:int)->Optional[ListNode]:
		dummy=ListNode(-1,head)
		fast=dummy
		slow=dummy
		n+=1
		while n>0:
			fast=fast.next
			n-=1
		while fast is not None:
			fast=fast.next
			slow=slow.next
		slow.next=slow.next.next
		return dummy.next

File: 卡码/代码随想录/1/02-链表/Go/05-删除链表倒数第n个节点/LeetCode_0019.go
//19. 删除链表的倒数第 N 个结点
package main

type ListNode struct{
	Val int
	Next *ListNode
}

func removeNthFromEnd(head *ListNode, n int) *ListNode {
	dummy:=&ListNode{-1,head}
	fast,slow:=dummy,dummy
	n++
	for n>0 {
		fast=fast.Next
		n--
	}
	for fast!=nil {
		fast=fast.Next
		slow=slow.Next
	}
	slow.Next=slow.Next.Next
	return dummy.Next
}
